services/supervisor/src/api.rs

```diff
@@ -51,7 +51,7 @@ pub trait PublicApi {
     /// Returns an actual consensus configuration of the blockchain.
     fn consensus_config(&self) -> Result<ConsensusConfig, Self::Error>;
     /// Returns an pending propose config change.
-    fn config_proposal(&self) -> Result<Option<ConfigProposalWithHash>, Self::Error>;
+    fn config_proposal(&self) -> Result<Vec<ConfigProposalWithHash>, Self::Error>;
```

Comment:
Also I think we still need a method to get a concrete proposal (either by height or by hash -- or even both).